### ISSUE TITLE
Code generation to JavaScript duplicates the input constants when they occur multiple times in the code

### ISSUE THREAD

**Issue by jeromesimeon** [querycert/qcert#83]
When the input query uses the same input constants multiple times, the generate code creates multiple bindings for it (but use only one). For instance:

```
bash-3.2$ ../bin/qcert -source oql -target javascript oql/union.oql
Compiling from oql to js:
  oql -> nraenv -> nraenv -> nnrc -> nnrc -> js
bash-3.2$ cat oql/union.js 
function query(constants) {
  var vCustomers_0 = deref(constants, "Customers");
  var vCustomers = deref(constants, "Customers");
  return concat({"c2": vCustomers}, {"c1": vCustomers});
}
```

**Comment by shinnar** [comment]
I believe that this is resolved by 4ae6a5f 
@jeromesimeon Please confirm and close.

**Comment by jeromesimeon** [comment]
Looks good! Thanks a lot.
